kythe/cxx/indexer/cxx/IndexerFrontendAction.cc

```diff
@@ -143,6 +145,15 @@ std::string IndexCompilationUnit(
     const LibrarySupports* LibrarySupports,
     std::function<std::unique_ptr<IndexerWorklist>(IndexerASTVisitor*)>
         CreateWorklist) {
+  llvm::sys::path::Style Style;
+  if (!kythe::IndexVFS::DetectStyleFromAbsoluteWorkingDirectory(
+          Unit.working_directory(), &Style)) {
```

Comment:
return `absl::optional<Style>` ? This then becomes:

llvm::sys::path::Style Style = Detect(...).value_or(Style::posix);

(You'd lose the warning on detection failure, but could move that into detect).